gateway/gateway-requests/src/registration/handshake/state.rs

```diff
@@ -15,9 +16,57 @@ use futures::{Sink, SinkExt, Stream, StreamExt};
 use log::*;
 use nymsphinx::params::{GatewayEncryptionAlgorithm, GatewaySharedKeyHkdfAlgorithm};
 use rand::{CryptoRng, RngCore};
+use serde::{Deserialize, Serialize};
 use std::convert::{TryFrom, TryInto};
 use tungstenite::Message as WsMessage;
 
+#[derive(Serialize, Deserialize)]
+pub struct InitMessage {
+    local_id_pubkey: Vec<u8>,
+    ephemeral_key: Vec<u8>,
+    credential: Option<Credential>,
+}
+
+impl InitMessage {
+    fn new(
+        local_id_pubkey: &identity::PublicKey,
+        ephemeral_key: &encryption::PublicKey,
+        credential: Credential,
+    ) -> Self {
+        InitMessage {
+            local_id_pubkey: local_id_pubkey.to_bytes().to_vec(),
+            ephemeral_key: ephemeral_key.to_bytes().to_vec(),
+            credential: Some(credential),
+        }
+    }
+
+    pub fn local_id_pubkey(&self) -> identity::PublicKey {
+        identity::PublicKey::from_bytes(&self.local_id_pubkey).unwrap()
```

Comment:
We definitely do not want to do unwraps here! We don't trust the data client is sending the gateway! They can send whatever bullshit they want and it could potentially crash the gateway if unwrap fails

Reply:
Though maybe it's fine because then deserialization would have failed... Hm... Could we perhaps just store the concrete identity::PublicKey on the struct and deal with the conversion during deserialization?